src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,495 @@
+								teacher: name,
+								class: classEntry.longname,
+								room: entry.ro[0].longname,
+								subject: entry.su[0].longname,
+								state: 'new',
+							});
+						}
+					}
+				}
+
+				// One teacher for now
+				break;
+			}
+		} else { // Iterate over rooms
+			for (const room of intermediateData.rooms) {
+				const name = `${room.name} (${room.longName}${room.building !== '' ? `, ${room.building}` : ''})`;
+
+				let timetable = await this.getCustomizableTimeTableFor(4, room.id, {
+					startDate: intermediateData.currentSchoolYear.startDate,
+					endDate: intermediateData.currentSchoolYear.endDate,
+					onlyBaseTimetable: true,
+					klasseFields: ['id', 'longname'],
+					subjectFields: ['id', 'longname'],
+					teacherFields: ['id', 'longname'],
+				});
+
+				/* TODO: Check for change */
+				timetable = timetable.filter(entry => entry.ro.length === 1
+					&& entry.ro[0].id === room.id
+					&& entry.kl.length > 0
+					&& entry.te.length === 1
+					&& entry.su.length === 1);
+				/* END TODO: Check for change */
+	
+				for (const entry of timetable) {
+					for (const classEntry of entry.kl) {
+						const klass = intermediateData.classes.find(k => k.id === classEntry.id);
+	
+						if (klass !== undefined) {
+							klass.timetable.push({
+								date: entry.date,
+								startTime: entry.startTime,
+								endTime: entry.endTime,
+								teacher: entry.te[0].longname,
+								subject: entry.su[0].longname,
+								room: name,
+							});
+
+							result.push({
+								teacher: entry.te[0].longname,
+								class: classEntry.longname,
+								room: entry.ro[0].longname,
+								subject: entry.su[0].longname,
+								state: 'new',
+							});
+						}
+					}
+				}
+
+				// One room for now
+				break;
+			}
+		}
+
+		/**
+		 * Result: Array of {
+		 * 	 teacher: 'Renz',
+		 * 	 class: '2a',
+		 * 	 room: '0-23',
+		 * 	 subject: 'mathe',
+		 * 	 state: 'new',
+		 * }
+		 */
+
+		return result;
+	}
+
+	/**
+	* 
+	* @param {*} session 
+	* @param {*} metaData 
+	*/
+	async fetchData(session, metaData) {
+		// Not required as all information are fetched during phase 1
+
+		return metaData;
+	}
+
+	/**
+	* 
+	* @param {*} session 
+	*/
+	async logout(session) {
+		await super.logout();
+	}
+
+	/**
+	* 
+	* @param {*} metadata 
+	*/
+	async storeMetadata(metaData, params) {
+		const webUntisMetadataService = this.app.service('webuntisMetadata');
+
+		await Promise.all(metaData.map(async (entry) => {
+			const metadataResults = await webUntisMetadataService.find({ query: {
+				datasourceId: params.datasourceId,
+				teacher: entry.teacher,
+				class: entry.class,
+				subject: entry.subject,
+				room: entry.room
+			}, paginate: false });
+			const result = metadataResults[0];
+
+			if (result !== undefined) { // patch/replace existing
+				await webUntisMetadataService.patch(result._id, { state: entry.state });
+			} else { // create new
+				await webUntisMetadataService.create(Object.assign(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use an object spread instead of `Object.assign` eg: `{ ...foo }`.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)